### PR TITLE
feat: Workspaces Page Query Filter

### PR DESCRIPTION
**Pull request by f0ssel**
Discussion at https://github.com/coder/coder/issues/1820

I wanted to get a proof of concept working so I could hand this off to a more skilled frontend engineer.

chrome_i0JLDryS2h

 I think the big points here I wanted to make are:
- Fast implementation
- Filter the data we already have, don't bother with backend filtering
- Have a sane default query value

### CODE REVIEW

**Comment by ammario** [comment]
Thanks for drafting this up.

Some notes:
* The search bar should be left aligned for consistency with the "Create Workspace" button
* I think we should deeply consider whether it's implemented on the frontend or the backend. In v1 we decided to put all list filtering and sorting in the frontend to the dismay of our mostly large customers. Then, we never got around to fixing it. We have a fast timeline to generate revenue with this product so expect to rebuild a frontend only implementation within a couple of months.

**Comment by f0ssel** [comment]
> Thanks for drafting this up.
> 
> Some notes:
> 
> * The search bar should be left aligned for consistency with the "Create Workspace" button
> * I think we should deeply consider whether it's implemented on the frontend or the backend. In v1 we decided to put all list filtering and sorting in the frontend to the dismay of our mostly large customers. Then, we never got around to fixing it. We have a fast timeline to generate revenue with this product so expect to rebuild a frontend only implementation within a couple of months.

1. Yeah I only cared about updating state, literally 0 time spent on visual. 
2. Totally, this is dead simple and would be very little to clean up if we wanted to move to backend filtering. I would expect a rework before multi-org since that would require new logic here to filter by orgs. If we wanted something by community MVP I would suggestion frontend only like I have here for velocity and effectiveness. Right now all the data for this table is fetched from a single http call.

**Comment by f0ssel** [comment]
I'm intentionally trying to get the most out of a impact / effort equation and being aware of the existing timelines and avoiding scope creep. This is something I can pair with someone to mostly implement myself if we want to protect the planned for the sprint to, I'm basically a pinch hitter right now.

File: site/src/pages/WorkspacesPage/WorkspacesPage.tsx [hunk 1]
```diff
@@ -1,16 +1,23 @@
...
+import { useMachine , useActor } from "@xstate/react"
+import React, { useContext } from "react"
 import { workspacesMachine } from "../../xServices/workspaces/workspacesXService"
 import { WorkspacesPageView } from "./WorkspacesPageView"
+import { XServiceContext } from "../../xServices/StateContext"
 
 const WorkspacesPage: React.FC = () => {
+  const xServices = useContext(XServiceContext)
+  const [authState] = useActor(xServices.authXService)
+  const { me } = authState.context
```
Comment (presleyp) [inline site/src/pages/WorkspacesPage/WorkspacesPage.tsx:10]: When you only need one thing out of an xservice, it's better to `useSelector`.

**Comment by f0ssel** [comment]
Nope really just wanted to illustrate the idea, not wanting this code merged

**Comment by misskniss** [comment]
This is going to be done via the backend with #1972

**Pull request closed without merging**